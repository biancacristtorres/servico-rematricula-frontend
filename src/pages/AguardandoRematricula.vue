<template>
  <v-container>
    <v-layout justify-center>
      <v-flex xs12 sm6 class="text-xs-center">
        <v-avatar :tile="true" :size="'96px'">
          <img :src="srcCapelo" alt="Capelo">
        </v-avatar>

        <span class="d-block grey--text text--darken-2 title mt-3">
          Agora é só esperar!
        </span>
      </v-flex>
    </v-layout>

    <v-layout justify-center>
      <v-flex xs12 sm8 class="text-xs-center grey--text text--darken-1 mt-4 subheading">
        <p>
          Durante a sua rematrícula, você terá acesso aos seus horários detalhadamente. 
          Estamos analisando as disciplinas que você demonstrou interesse.
        </p>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script lang='ts'>
import { Vue, Prop } from 'vue-property-decorator';
import { Component, Mixin, Mixins } from 'vue-mixin-decorator';
import { ChatMixin } from '@/mixin/ChatMixin';

@Component({})
export default class AguardandoRematricula extends Mixins<ChatMixin>(ChatMixin) {
  public get srcCapelo() {
    return `${process.env.BASE_URL}svg/capelo.svg`;
  }
}
</script>