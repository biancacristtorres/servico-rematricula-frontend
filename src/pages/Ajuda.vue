<template>
<v-container bg fill-height grid-list-md text-xs-center>
  <v-layout row wrap align-center>
    <v-flex class="display-1 grey--text">
      Redirecionando...
    </v-flex>
  </v-layout>
</v-container>
</template>

<script lang='ts'>
import { ObterAcessoAoFreshDeskService } from '@/services/ObterAcessoAoFreshDeskService';
import { Vue, Component } from 'vue-property-decorator';
import { StoreNamespaces } from '@/store';
import { namespace } from 'vuex-class';
import { Container } from 'typescript-ioc';
import UsuarioIdentity from 'componente-frontend-core/model/UsuarioIdentity';

const autorizacaoNamespace = namespace(StoreNamespaces.AUTORIZACAO);

@Component({})
export default class Ajuda extends Vue {
  @autorizacaoNamespace.State
  public usuarioLogado!: UsuarioIdentity;

  private get obterAcessoAoFreshDeskService(): ObterAcessoAoFreshDeskService {
    return Container.get(ObterAcessoAoFreshDeskService);
  }

  public async mounted() {
    this.obterAcessoAoFreshDeskService.processar(this.usuarioLogado.codigo || 0)
      .then((response) => {
        location.href = response;
      });
  }
}
</script>