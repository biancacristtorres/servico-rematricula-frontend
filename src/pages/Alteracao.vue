
<template>
  <v-container class="pa-0">
    <v-layout row wrap class="pa-0">
      <v-flex xs12 md8 class="px-0 pt-4">
        <v-system-bar>
          <v-toolbar-title class="subheading font-weight-bold">
            Disciplinas dispon√≠veis
          </v-toolbar-title>

          <v-spacer></v-spacer>

          <v-toolbar-items v-if="deveExibirOFiltroDeTurmas" 
            id="filtro-de-turmas" 
            class="hidden-sm-and-down">

            <filtro-de-turma>
            </filtro-de-turma>
          </v-toolbar-items>
        </v-system-bar>
      </v-flex>

      <v-flex xs12 md8>
        <disciplinas-disponiveis>
        </disciplinas-disponiveis>
      </v-flex>

      <v-flex xs12 md4>
        <v-container class="px-2">
          <salvar-alteracoes>
          </salvar-alteracoes>

          <mini-grade-de-horario>
          </mini-grade-de-horario>
        </v-container>
      </v-flex>

    </v-layout>
  </v-container>
</template>

<script lang='ts'>
import { Vue, Prop } from 'vue-property-decorator';
import { Component, Mixin, Mixins } from 'vue-mixin-decorator';
import { Container } from 'typescript-ioc';
import { ChatMixin } from '@/mixin/ChatMixin';
import { ToggleConstants } from '@/common/enum/ToggleConstants';
import { StoreNamespaces } from '@/store';
import { namespace, Action } from 'vuex-class';
import { FeatureToggleGetterTypes } from 'componente-frontend-core/store/featureToggle/getters';
import { AlteracaoActionTypes } from '@/store/alteracao/actions';
import { MiniGradeActionTypes } from '@/store/miniGrade/actions';
import { GlobalActionTypes } from '@/store/actions';
import { HorarioDaMiniGrade } from '@/model/HorarioDaMiniGrade';
import {
  PreparacaoDeTurmasService,
} from '@/services/alteracao/PreparacaoDeTurmasService';
import {
  AcessarAlteracaoService,
} from '@/services/alteracao/AcessarAlteracaoService';
import store from '@/store';
import UsuarioIdentity from 'componente-frontend-core/model/UsuarioIdentity';
import DisciplinasDisponiveis from '@/components/alteracao/DisciplinasDisponiveis.vue';
import SalvarAlteracoes from '@/components/alteracao/SalvarAlteracoes.vue';
import MiniGradeDeHorario from '@/components/alteracao/MiniGradeDeHorario.vue';
import FiltroDeTurma from '@/components/alteracao/FiltroDeTurma.vue';

const autorizacao = namespace(StoreNamespaces.AUTORIZACAO);
const alteracao = namespace(StoreNamespaces.ALTERACAO);
const miniGrade = namespace(StoreNamespaces.MINI_GRADE);

@Component({
  components: {
    DisciplinasDisponiveis,
    SalvarAlteracoes,
    MiniGradeDeHorario,
    FiltroDeTurma,
  },
})
export default class Alteracao extends Mixins<ChatMixin>(ChatMixin) {

  private get codigoDoAluno(): number {
    return this.usuarioLogado.codigo || 0;
  }

  private get preparacaoDeTurmasService(): PreparacaoDeTurmasService {
    return Container.get(PreparacaoDeTurmasService);
  }

  public deveExibirOFiltroDeTurmas: boolean = false;

  @autorizacao.State
  public usuarioLogado!: UsuarioIdentity;

  @alteracao.Action(AlteracaoActionTypes.PREPARANDO_TURMAS)
  public preparandoTurmas!: () => Promise<void>;

  @alteracao.Action(AlteracaoActionTypes.PREPARACAO_FINALIZADA)
  public preparacaoFinalizada!: () => Promise<void>;

  @miniGrade.Action(MiniGradeActionTypes.CARREGAR_MINI_GRADE)
  public carregarMiniGrade!: (horarios: HorarioDaMiniGrade[]) => Promise<void>;

  @Action(GlobalActionTypes.DEFINIR_LINK_PARA_VOLTAR)
  public definirLinkParaVoltar!: (link: string) => Promise<void>;

  @alteracao.Action(AlteracaoActionTypes.LIMPAR_ALTERACAO)
  private limparAlteracoes!: () => Promise<void>;

  public async beforeRouteEnter(to: any, from: any, next: any) {
    const codigoDoAluno = (store.state as any).autorizacao.usuarioLogado.codigo;

    try {
      const response = await Container.get(AcessarAlteracaoService).processar(codigoDoAluno);

      if (response.data.statusEstaLiberadoParaAlteracao) {
        next();
      } else {
        next('/Resumo');
      }
    } catch {
      next('/Resumo');
    }
  }

  public async created() {
    await this.definirLinkParaVoltar('/Resumo');
    await this.limparAlteracoes();
    await this.setarExibicaoDoFiltroDeTurmas();

    this.preparandoTurmas();
    await this.prepararTurmas();
    this.preparacaoFinalizada();
  }


  public async setarExibicaoDoFiltroDeTurmas(): Promise<void> {
    this.deveExibirOFiltroDeTurmas = await this.$features
      .habilitado(ToggleConstants.HabilitarFiltroDeTurmas);
  }

  private async prepararTurmas(): Promise<void> {
    const response = await this.preparacaoDeTurmasService.processar(this.codigoDoAluno);
    this.carregarMiniGrade(response.data.horarios);
  }

}
</script>