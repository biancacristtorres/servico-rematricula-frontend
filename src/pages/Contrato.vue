<template>
  <div>
    <v-container v-show="podeExibirMensagemDeCarregamento">
      <v-layout>
        <v-flex xs12 class="text-xs-center">
          <v-progress-circular class="mt-5" :size="70" :width="7" color="primary" indeterminate>
          </v-progress-circular>
        </v-flex>
      </v-layout>
    </v-container>

    <div v-show="podeExibirContrato">
      <v-container>
        <v-layout class="container-pdf">
          <v-flex xs12>
            <arquivo-de-contrato @download-pronto="aoFinalizarDownload"></arquivo-de-contrato>
          </v-flex>
        </v-layout>
      </v-container>

      <v-footer height="80" app fixed class="justify-center elevation-10">
        <assinar-contrato></assinar-contrato>
      </v-footer>
  </div>
  </div>
</template>

<script lang='ts'>
import { Vue, Prop } from 'vue-property-decorator';
import { Component, Mixin, Mixins } from 'vue-mixin-decorator';
import { ChatMixin } from '@/mixin/ChatMixin';

import ArquivoDeContrato from '@/components/contrato/ArquivoDeContrato.vue';
import AssinarContrato from '@/components/contrato/AssinarContrato.vue';

@Component({
  components: {
    ArquivoDeContrato,
    AssinarContrato,
  },
})
export default class Contrato extends Mixins<ChatMixin>(ChatMixin) {
  public carregando: boolean = true;

  public aoFinalizarDownload(): void {
    this.carregando = false;
  }

  private get podeExibirMensagemDeCarregamento(): boolean {
    return this.carregando;
  }

  private get podeExibirContrato(): boolean {
    return !this.carregando;
  }
}
</script>

<style lang="stylus" scoped>
@import '~@/common/stylus/variaveis.styl'
tamanho-footer = 120px
calculo-heigth-por-tamanho-dispositivo = "calc(100vh - %s - " + tamanho-footer + ")"

@media dispositivo-pequeno
  .container-pdf
    overflow-y auto
    height: calculo-heigth-por-tamanho-dispositivo % tamanho-toolbar-dispositivo-pequeno
    
@media dispositivo-medio
  .container-pdf
    height: calculo-heigth-por-tamanho-dispositivo % tamanho-toolbar-dispositivo-medio

@media dispositivo-grande
  .container-pdf
    height: calculo-heigth-por-tamanho-dispositivo % tamanho-toolbar-dispositivo-grande
</style>
