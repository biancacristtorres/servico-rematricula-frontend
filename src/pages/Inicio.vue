<template>
  <v-container>
    <v-layout justify-center class="hidden-md-and-up">
      <v-flex xs12 sm8 class="text-xs-center grey--text text--darken-1 mt-4 subheading">
        <iframe style="max-width: 100%" src="https://www.youtube.com/embed/d-mW4beRnoo" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen>
        </iframe>
      </v-flex>
    </v-layout>

    <v-layout justify-center>
      <v-flex xs12 sm6 class="text-xs-center">
        <v-avatar :tile="true" :size="'96px'">
          <img :src="srcCapelo" alt="Capelo">
        </v-avatar>

        <span class="d-block grey--text text--darken-2 title mt-3">Novidades =)</span>
      </v-flex>
    </v-layout>

    <v-layout justify-center class="hidden-sm-and-down">
      <v-flex xs12 sm8 class="text-xs-center grey--text text--darken-1 mt-4 subheading">
        <iframe width="560" height="315" src="https://www.youtube.com/embed/d-mW4beRnoo" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen>
        </iframe>
      </v-flex>
    </v-layout>

    <v-layout justify-center class="mt-4">
      <v-flex xs12 sm6>
        <v-btn id="continuar" @click.native="redirecionar" block large color="primary">Continuar</v-btn>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script lang='ts'>
import { Component, Vue } from 'vue-property-decorator';
import { ToggleConstants } from '@/common/enum/ToggleConstants';

@Component({})
export default class Inicio extends Vue {

  public async redirecionar() {
    const deveExibirInteresse = await this.$features
      .habilitado(ToggleConstants.HabilitarInteresseDeDisciplinas);

    if (deveExibirInteresse) {
      this.$router.push('/Interesse');
    } else {
      this.$router.push('/AguardandoRematricula');
    }
  }

  public get srcCapelo() {
    return `${process.env.BASE_URL}svg/capelo.svg`;
  }

}
</script>