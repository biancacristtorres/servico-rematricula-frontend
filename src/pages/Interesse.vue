<template>
  <v-container>
    <v-layout justify-center>
      <v-flex xs12 sm6 class="text-xs-center">
        <v-avatar :tile="true" :size="'96px'">
          <img :src="srcWarning" alt="Alerta">
        </v-avatar>

        <span class="d-block grey--text text--darken-2 title">
          Disciplinas pendentes
        </span>
      </v-flex>
    </v-layout>

    <v-layout justify-center>
      <v-flex xs12 sm8 class="text-xs-center grey--text text--darken-1 mt-3 subheading">
        Atenção. Este interesse não garante vagas no seu próximo módulo,
        porém nos ajudará a buscar vagas nestas disciplinas.
      </v-flex>
    </v-layout>

    <v-layout justify-center>
      <v-flex xs12 sm6>
        <selecao-de-disciplinas 
          @interesseSolicitado="aoSolicitarInteresse" 
          @alertaEmitido="aoEmitirAlerta">
        </selecao-de-disciplinas>

        <selecao-de-condicao-especial 
          @condicaoSolicitada="aoSolicitarCondicao"
          :quantidadeDeDisciplinasComInteresse="quantidadeDeDisciplinasComInteresse">
        </selecao-de-condicao-especial>

        <alerta-de-regras :exibirAlertaDeRegras="exibirAlertaDeRegras">
        </alerta-de-regras>

        <continuar :condicoes="condicoes" :disciplinas="disciplinas">
        </continuar>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script lang='ts'>
import {
  Component,
  Prop,
  Vue,
} from 'vue-property-decorator';
import {
  DisciplinaComInteresse,
} from '@/model/DisciplinaComInteresse';
import {
  CondicaoEspecial,
} from '@/common/enum/CondicaoEspecial';
import SelecaoDeDisciplinas from '@/components/interesse/SelecaoDeDisciplinas.vue';
import SelecaoDeCondicaoEspecial from '@/components/interesse/SelecaoDeCondicaoEspecial.vue';
import AlertaDeRegras from '@/components/interesse/AlertaDeRegras.vue';
import Continuar from '@/components/interesse/Continuar.vue';

@Component({
  components: {
    SelecaoDeDisciplinas,
    SelecaoDeCondicaoEspecial,
    AlertaDeRegras,
    Continuar,
  },
})
export default class Interesse extends Vue {
  public condicoes: CondicaoEspecial[] = [];
  public disciplinas: DisciplinaComInteresse[] = [];
  public exibirAlertaDeRegras: boolean = false;

  public get srcWarning() {
    return `${process.env.BASE_URL}svg/warning.svg`;
  }

  public aoSolicitarCondicao(condicoes: CondicaoEspecial[]) {
    this.condicoes = condicoes;
  }

  public aoSolicitarInteresse(disciplinas: DisciplinaComInteresse[]) {
    this.disciplinas = disciplinas;
  }

  private get quantidadeDeDisciplinasComInteresse(): number {
    return this.disciplinas.length;
  }

  public aoEmitirAlerta() {
    this.exibirAlertaDeRegras = true;
  }
}
</script>