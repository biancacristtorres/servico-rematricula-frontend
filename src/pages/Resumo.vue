<template>
  <v-container>
    <v-layout row wrap align-center>
      <v-flex class="text-xs-center">
        <acoes></acoes>
        <grade-de-horario></grade-de-horario>
        <disciplinas-sem-horario></disciplinas-sem-horario>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script lang='ts'>

import { Vue } from 'vue-property-decorator';
import { Component, Mixin, Mixins } from 'vue-mixin-decorator';
import { ChatMixin } from '@/mixin/ChatMixin';

import GradeDeHorario from '@/components/resumo/GradeDeHorario.vue';
import Acoes from '@/components/resumo/Acoes.vue';
import DisciplinasSemHorario from '@/components/resumo/DisciplinasSemHorario.vue';

@Component({
  components: {
    GradeDeHorario,
    Acoes,
    DisciplinasSemHorario,
  },
})

export default class Resumo extends Mixins<ChatMixin>(ChatMixin) {
}

</script>