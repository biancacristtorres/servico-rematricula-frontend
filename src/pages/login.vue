<template>
  <v-container fluid fill-height grid-list-xl>
    <v-layout justify-center align-center>
      <v-flex lg4 md6 sm8 xs12>
        <v-card class="pa-5" raised>
          <v-form>
            <v-alert
              :value="possuiErroCredenciais"
              transition="scale-transition" 
              color="error"
              class="my-4"
              icon="warning"
              outline>
              O R.A. ou senha preenchidos n√£o foram encontrados no sistema. Favor corrigir e tentar novamente.
            </v-alert>

            <v-text-field 
              @keyup.enter="entrar" 
              :error="erros.usuario" 
              data-vv-name="usuario" 
              v-validate="'required'"
              v-model="login"
              name="usuario" 
              label="R.A."
              autocomplete="username" 
              id="login-usuario">
            </v-text-field>

            <v-text-field 
              @keyup.enter="entrar" 
              :error="erros.senha" 
              data-vv-name="senha"
              v-validate="'required'"
              v-model="senha"
              name="senha"
              label="Senha"
              type="password"
              autocomplete="current-password"
              id="login-senha">
            </v-text-field>

            <v-card-actions class="mt-4 justify-center pa-0">
              <v-btn :loading="loading" class="primary" block large @click="entrar" id="login-botao">
                Entrar
              </v-btn>
            </v-card-actions>
          </v-form>
        </v-card>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script lang="ts">
import {
  Component,
  Prop,
  Vue,
} from 'vue-property-decorator';

import { namespace } from 'vuex-class';
import { StoreNamespaces } from '@/store';
import { AutorizacaoActionTypes } from 'componente-frontend-core/store/autorizacao/actions';

const autorizacaoNamespace = namespace(StoreNamespaces.AUTORIZACAO);

@Component({})
  export default class Login extends Vue {
    public login: string = '';
    public senha: string = '';
    public possuiErroCredenciais = false;
    public loading = false;
    public erros: any = { usuario: false, senha: false };

    @autorizacaoNamespace.Action(AutorizacaoActionTypes.FAZER_LOGIN)
    public fazerLogin!: (dados: any) => Promise<string>;

    private entrar(): void {
      this.resetarEstadosDeErro();

      this.$validator
        .validateAll()
        .then(this.aoValidarFormulario);
    }

    private resetarEstadosDeErro(): void {
      this.possuiErroCredenciais = false;
      this.erros = { usuario: false, senha: false };
    }

    private aoValidarFormulario(estaValido: any): void {
      if (estaValido) {
        this.quandoOFormularioEstaValido();
      } else {
        this.quandoOFormularioEstaInvalido();
      }
    }

    private quandoOFormularioEstaValido(): void {
      this.ligarLoading();
      this.fazerLogin({ usuario: this.login, senha: this.senha })
        .then(this.loginComSucesso)
        .catch(this.falhaNoLogin)
        .finally(this.desligarLoading);
    }

    private quandoOFormularioEstaInvalido(): void {
        this.erros = { usuario: this.errors.has('usuario'), senha: this.errors.has('senha') };
    }

    private loginComSucesso(): void {
      this.$router.push('/');
    }

    private falhaNoLogin(): void {
      this.possuiErroCredenciais = true;
    }

    private ligarLoading() {
      this.loading = true;
    }

    private desligarLoading() {
      this.loading = false;
    }
}
</script>

