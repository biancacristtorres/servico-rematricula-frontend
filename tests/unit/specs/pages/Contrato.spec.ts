import { ContratoFactory } from '../../factory/pages/ContratoFactory';

describe('PÃ¡gina de Contrato', () => {
  const factory = new ContratoFactory();

  it('Snapshot', async (done) => {
    await factory.montarComponente();
    expect(factory.componente).toMatchSnapshot();
    done();
  });

});
