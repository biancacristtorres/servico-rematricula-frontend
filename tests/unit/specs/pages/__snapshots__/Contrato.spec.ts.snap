// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PÃ¡gina de Contrato Snapshot 1`] = `
<div>
  <vcontainer-stub tag="div">
    <vlayout-stub tag="div">
      <vflex-stub tag="div" xs12="" class="text-xs-center">
        <vprogresscircular-stub indeterminate="true" rotate="0" size="70" width="7" value="0" color="primary" class="mt-5"></vprogresscircular-stub>
      </vflex-stub>
    </vlayout-stub>
  </vcontainer-stub>
  <div style="display: none;">
    <vcontainer-stub tag="div">
      <vlayout-stub tag="div" class="container-pdf">
        <vflex-stub tag="div" xs12="">
          <arquivodecontrato-stub></arquivodecontrato-stub>
        </vflex-stub>
      </vlayout-stub>
    </vcontainer-stub>
    <vfooter-stub height="80" app="" fixed="" class="justify-center elevation-10">
      <assinarcontrato-stub></assinarcontrato-stub>
    </vfooter-stub>
  </div>
</div>
`;
